enumerable.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i], i)
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      (0..length - 1).my_each do |i|
+        new_arr.push(self[i]) if yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_all?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def my_any?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def my_none?
+    i = 0
+    while i < size
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count
+    i = 0
+    count = []
+    if block_given?
+      while i < size
+        count << self[i] if yield(self[i])
+        i += 1
+      end
+      return count.size
+    end
+    size
+  end
+
+  def my_map(&block)
+    arr = []
+    my_each do |e|
+      arr << block.call(e)
+    end
+    arr
+  end
```

Comment:
- We need to fix my_each first to make this method works as expected.

- returns a new array with the results of running block once for every element in enum.
`[1,2,3,4,5].map { |x| x + 5 }`
`=> [6, 7, 8, 9, 10]`

- My_map returns an Enumerator if no block is given
`[1,2,3,4,5].map`
`=> #<Enumerator: [1, 2, 3, 4, 5]:map>`